styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -133,34 +136,36 @@
   private final Debug debug;
   private final int pollPodsIntervalSeconds;
   private final int podDeletionDelaySeconds;
-  private final Clock clock;
+  private final Time time;
+  private final ConcurrentMap<String, Long> submissions = new ConcurrentHashMap<>();
```

Comment:
Hard to read, have to guess the semantics until I read into usage. `submissionTimeByExecutionId` or a comment saying as much would be superb.